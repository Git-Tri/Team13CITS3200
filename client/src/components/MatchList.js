import React, { Component } from 'react';
import PageHeader from './PageHeader.js';
import { Container, Form, Input } from 'semantic-ui-react'


class MatchList extends Component {

	state = {
		name: '',
		startdate: 0,
		enddate: 0,
		league: ''
	}

	submitHandler = (e) => {
		e.preventDefault()
		console.log(this.state)
		var request = new XMLHttpRequest()
		request.open('GET', '/')
	}

	render() {
		const { name, startdate, enddate, league } = this.state
		return (
			<div className="page">
				<PageHeader 
					header={"Match List"}
					sidebarVisible={this.props.sidebarVisible}
					handleSidebarClick={this.props.handleSidebarClick}
				/>
				<Container style={{height:"100vh"}}>
					<Form>
						<Form.Field>
							<Input label="Search" type="text" name="search-name"/>
						</Form.Field>
						<Form.Group widths="equal">
							<Form.Field>
								<Input label="Between" type="date" name="search-startdate" placeholder="Start date"/>
							</Form.Field>
							<Form.Field>
								<Input type="date" name="search-enddate" placeholder="End date"/>
							</Form.Field>
							<Form.Field>
								<Input label="League" type="text" name="search-league"/>
							</Form.Field>
						</Form.Group>
						<Form.Button onClick="submitHandler">Submit</Form.Button>
					</Form>

					<div class="results"></div>
				</Container>
			</div>
		);
	}
}

export default MatchList;
