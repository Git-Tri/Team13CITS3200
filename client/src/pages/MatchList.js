import React, { Component } from 'react';
import { Header } from 'semantic-ui-react';

class MatchList extends Component {
	render() {
		return (
			<div className="page">
				<Header>
					Match List
				</Header>
			</div>
		);
	}
}

export default MatchList;