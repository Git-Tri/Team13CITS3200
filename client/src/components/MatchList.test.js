import MatchList from './MatchList';
import renderer from 'react-test-renderer';
import React from 'react';
import {shallow, mount, configure} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from "enzyme-to-json";

configure({ adapter: new Adapter() });

describe("Match List Tests",function() {
	let page;
	let mockSubmit;

	beforeEach(() => {
		mockSubmit = jest.fn();
		page = shallow(<MatchList submit={mockSubmit} />);
	});

	it("should match snapshot",() => {
		expect(page).toMatchSnapshot();
	})

	it("should call handleChange on searchtext", () => {
		const mockEvent = {
			target: {
				name: 'searchtext',
				value: "Sample text"
			}
		};
		const expected = {
			searchtext: "Sample text",
			startdate: '0000-00-00',
			enddate: '0000-00-00',
			league: '',
			dateError: false
		};
		page.instance().handleChange(mockEvent);
		expect(page.state()).toEqual(expected);
	});

	it("should throw error if dates are invalid", () => {
		page.setState({'startdate':'2001-01-02'})
		page.setState({'enddate':'2001-01-01'})
		expect(page.state().dateError).toEqual(true)
	});
})
