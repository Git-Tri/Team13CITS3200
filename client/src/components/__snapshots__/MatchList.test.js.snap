// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Match List Tests should match snapshot 1`] = `
<div
  className="page"
>
  <PageHeader
    backButtonRequired={true}
    header="Match List"
  />
  <Container
    style={
      Object {
        "height": "100vh",
      }
    }
  >
    <Form
      as="form"
    >
      <FormField>
        <Input
          label="Search"
          name="searchtext"
          onChange={[Function]}
          type="text"
        />
      </FormField>
      <FormGroup
        widths="equal"
      >
        <FormField>
          <Input
            label="Between"
            name="startdate"
            onChange={[Function]}
            placeholder="Start date"
            type="date"
          />
        </FormField>
        <FormField>
          <Input
            name="enddate"
            onChange={[Function]}
            placeholder="End date"
            type="date"
          />
        </FormField>
        <FormField>
          <Input
            label="Competition"
            name="competition"
            onChange={[Function]}
            type="text"
          />
        </FormField>
      </FormGroup>
      <FormButton
        as={[Function]}
        control={[Function]}
        onClick={[Function]}
      >
        Submit
      </FormButton>
    </Form>
    <div
      id="container"
      style={
        Object {
          "height": "100vh",
        }
      }
    >
      <MatchListTable
        items={
          Array [
            Match {
              "away": "team B",
              "competitionID": 1,
              "competitionName": "comp A",
              "date": 1991-04-20T00:00:00.000Z,
              "home": "team A",
              "id": 1,
            },
            Match {
              "away": "test D",
              "competitionID": 1,
              "competitionName": "comp B",
              "date": 1991-04-20T00:00:00.000Z,
              "home": "team C",
              "id": 2,
            },
          ]
        }
      />
    </div>
  </Container>
</div>
`;
